### DISCUSSION TITLE
Unable to load contents of file list in fastlane snapshot

### DISCUSSION THREAD

**Discussion by nyan-lin-tun** [fastlane/fastlane#17689] [Q&A]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

After upgrading to Xcode 12.2, snapshot is not working as expected. 
I had tried ``` pod deintegrate ``` and move dir from input/output file lists to input/output files from [this](https://stackoverflow.com/questions/55011171/xcode-buildtime-error-unable-to-load-contents-of-file-list-info-plist)
Error logs are same when try to run "fastlane snapshot". Please kindly see the error logs below.

``` 
03:05:12]: ▸ 2020-11-28 03:05:12.306 xcodebuild[15714:1311364] Error writing xctestrun file: Error Domain=NSCocoaErrorDomain Code=4 "The folder “FastlaneSnapshots_iphonesimulator14.2-arm64-x86_64.xctestrun” doesn’t exist." UserInfo={NSFilePath=/var/folders/2x/htb5g5n939b41nf983j2s1fw0000gn/T/snapshot_derived20201128-13026-2nutdl/Build/Products/FastlaneSnapshots_iphonesimulator14.2-arm64-x86_64.xctestrun, NSUserStringVariant=Folder, NSUnderlyingError=0x7f95a7e469d0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
[03:05:12]: ▸ ** TEST INTERRUPTED **

❌  error: Unable to load contents of file list: '/Target Support Files/Realm/Realm-xcframeworks-input-files.xcfilelist' (in target 'Realm' from project 'Pods')



❌  error: Unable to load contents of file list: '/Target Support Files/Realm/Realm-xcframeworks-output-files.xcfilelist' (in target 'Realm' from project 'Pods')


    Unable to read contents of XCFileList '/Target Support Files/Realm/Realm-xcframeworks-output-files.xcfilelist' (in target 'Realm' from project 'Pods')
** BUILD FAILED **
```
When testing FastlaneSnapshots in Xcode, it show the following error,

```
ld: file not found: /Users/nyanlintun/Library/Developer/Xcode/DerivedData/Project-dyswnxaltxrshveaoodsmmmnilfr/Build/Products/Production-iphonesimulator/Project Develop.app/Project Develop
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 11.0.1                                      |
| Ruby                        | 2.7.0                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.24.3 (Apple Git-128)          |
| Installation Source         | ~/.rvm/gems/ruby-2.7.0/bin/fastlane         |
| Host                        | macOS 11.0.1 (20B29)                        |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.7.0/lib                |
| OpenSSL Version             | OpenSSL 1.1.1g  21 Apr 2020                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 12.2                                        |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:ios)

platform :ios do
  lane :register_app do
    produce(
        username: '',
        app_identifier: '',
        app_name: 'Project Name',
        language: 'English',
        team_name: '',
        itc_team_name: ''
    )
  end

  lane :get_dev_certs do
    cert development: true
    sigh development: true
  end

  desc "Update iOS UDID's on the Developer Portal"
  lane :sync_device_info do
    register_devices(
      devices_file: "fastlane/Devicefile"
    )
  end

  desc "Sync team code-signing asstes"
  lane :sync_signing_assets do |options|
    sync_device_info
    selectedType = options[:type]
    match(type: selectedType)
  end

  # for manual match process
  lane :bootstrap_code_signing do
    sync_device_info
    match(type: 'development')
    match(type: 'adhoc')
    match(type: 'appstore')
  end

  desc "Build for app Store submission"
  lane :build_appstore do
    sync_signing_assets(type: "appstore")
    # syn team appstore creds
    increment_build_number
    gym(
      output_directory: "build_Appstore",
      export_method: "app-store"
    )
  end

  

  desc "Build for Ad Hoc submission"
  lane :build_adhoc do
    sync_signing_assets(type: "adhoc")
    # syn team appstore creds
    increment_build_number
    gym(
      output_directory: "build_AdHoc",
      export_method: "ad-hoc"
    )
  end

  lane :distribute_to_appstore do
    build_appstore
    pilot(
      team_name: "Team Name",
      changelog: "Version #{lane_context[SharedValues::VERSION_NUMBER]}, Build #{lane_context[SharedValues::BUILD_NUMBER]}"
    )
  end

  lane :distribute_to_installr do
    build_adhoc
    installr(
      api_token: "",
      notes: ""
    )
  end

  lane :sandbox do
    # cert development: true
    # sigh development: true
    # produce
    frameit(path: "./fastlane/screenshots")
    deliver(
      ipa: "./build_AppStore/ProjectName.ipa",
      force: true,
      submit_for_review: true
    )
  end

  lane :screenshots_with_frames do
    frameit(path: "./fastlane/screenshots", rose_gold: true)
  end

  lane :do_snapshot do
    snapshot
  end

  lane :release do
    precheck
    build_appstore
    snapshot
    # frameit(path: "./fastlane/screenshots", rose_gold: true)
    deliver(
      ipa: "./build_AppStore/ProjectName.ipa",
      run_precheck_before_submit: false,
      submit_for_review: true,
      automatic_release: true,
      force: true,
      submission_information: {
        add_id_info_limits_tracking: true,
        add_id_info_serves_ads: false,
        add_id_info_tracks_action: true,
        add_id_info_tracks_install: true,
        add_id_info_uses_idfa: true,
        content_rights_has_rights: true,
        content_rights_contains_third_party_content: true,
        export_compliance_platform: 'ios',
        export_compliance_compliance_required: false,
        export_compliance_encryption_updated: false,
        export_compliance_app_type: nil,
        export_compliance_uses_encryption: false,
        export_compliance_is_exempt: false,
        export_compliance_contains_third_party_cryptography: false,
        export_compliance_contains_proprietary_cryptography: false,
        export_compliance_available_on_french_store: false
      },
      team_name: "Team Name"
    )
  end

  lane :distribute_to_testflight do
    build_appstore
    pilot(
      username: "email",
      app_identifier: "com.bundleidentifier",
      wait_processing_interval: 30,
      distribute_external: true,
      skip_submission: true, #allows build script to end before processing the binary
      team_name: "Team Name",
      changelog: "Version #{lane_context[SharedValues::VERSION_NUMBER]}, Build #{lane_context[SharedValues::BUILD_NUMBER]}"
    )
  end

  lane :build_staging do
    sigh(app_identifier: "com.bundleidentifier.staging")
    sync_signing_assets(type: "appstore")
    increment_build_number
    gym(
      include_bitcode: false,
      output_directory: "build_AdHoc",
      export_method: "app-store"
    )
  end

  lane :distribute_staging_to_testflight do
    build_staging
    pilot(
      username: "email",
      app_identifier: "com.bundleidentifier.staging",
      wait_processing_interval: 30,
      distribute_external: true,
      skip_submission: true, #allows build script to end before processing the binary
      team_name: "Team Name", 
      changelog: "ChangeLog",
      ipa: "./build_AdHoc/Project.ipa" #path of the ipa generated by your build script
    )
  end

  lane :do_deliver do
    deliver(
      # ipa: "./build_AppStore/Project.ipa",
      build_number: "21",
      run_precheck_before_submit: false,
      submit_for_review: true,
      automatic_release: true,
      force: true,
      submission_information: {
        add_id_info_limits_tracking: true,
        add_id_info_serves_ads: false,
        add_id_info_tracks_action: true,
        add_id_info_tracks_install: true,
        add_id_info_uses_idfa: true,
        content_rights_has_rights: true,
        content_rights_contains_third_party_content: true,
        export_compliance_platform: 'ios',
        export_compliance_compliance_required: false,
        export_compliance_encryption_updated: false,
        export_compliance_app_type: nil,
        export_compliance_uses_encryption: false,
        export_compliance_is_exempt: false,
        export_compliance_contains_third_party_cryptography: false,
        export_compliance_contains_proprietary_cryptography: false,
        export_compliance_available_on_french_store: false
      },
      team_name: "Team Name"
    )
  end

  
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier ENV['SCHEME']
apple_id "email" 
team_name "Team Name"
team_id ""
itc_team_name "Team Name"
itc_team_id ""
ENV["FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD"] = ""
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.168.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.4.0        |
| executable-hooks          | 1.6.0        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.5        |
| rake                      | 13.0.1       |
| CFPropertyList            | 3.0.2        |
| public_suffix             | 4.0.6        |
| addressable               | 2.7.0        |
| atomos                    | 0.1.3        |
| aws-eventstream           | 1.1.0        |
| aws-partitions            | 1.399.0      |
| aws-sigv4                 | 1.2.2        |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.109.3      |
| aws-sdk-kms               | 1.39.0       |
| aws-sdk-s3                | 1.85.0       |
| babosa                    | 1.0.4        |
| bundler                   | 2.1.2        |
| claide                    | 1.0.3        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.6.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.6        |
| emoji_regex               | 3.2.1        |
| excon                     | 0.78.0       |
| multipart-post            | 2.0.0        |
| ruby2_keywords            | 0.0.2        |
| faraday                   | 1.1.0        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.7        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.2.2        |
| memoist                   | 0.16.2       |
| multi_json                | 1.15.0       |
| os                        | 1.1.1        |
| signet                    | 0.14.0       |
| googleauth                | 0.14.0       |
| httpclient                | 2.8.3        |
| mini_mime                 | 1.0.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.4.0        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.29.1       |
| json                      | 2.3.1        |
| mini_magick               | 4.11.0       |
| plist                     | 3.5.0        |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.3.0        |
| xcodeproj                 | 1.19.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2020-11-28**
</details>

**Comment by hjh5696** [comment]
same here. Any workaround?

**Reply by asam139** [reply]
same too :/

**Comment by sanduluca** [comment]
Hi. I have same problem. Did someone find a fix ? 
Fastlane version 2.212.2

Edit: Managed to solve my issue. The actual error was 1 step before in our CI. The `pod install` was failing. I had to add the `--repo-update` flag so that the pod install step could finish successfully.
